system/micd.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+import sounddevice as sd
+import numpy as np
+
+from cereal import messaging
+from common.filter_simple import FirstOrderFilter
+from common.realtime import Ratekeeper
+from system.swaglog import cloudlog
+
+RATE = 10
+DT_MIC = 1. / RATE
+REFERENCE_SPL = 2 * 10 ** -5  # newtons/m^2
+
+
+class Mic:
+  def __init__(self, pm):
+    self.pm = pm
+    self.rk = Ratekeeper(RATE)
+
+    self.measurements = np.empty(0)
+    self.spl_filter = FirstOrderFilter(0, 4, DT_MIC, initialized=False)
+
+  def update(self):
+    # self.measurements contains amplitudes from -1 to 1 which we use to
+    # calculate an uncalibrated sound pressure level
+    if len(self.measurements) > 0:
+      # https://www.engineeringtoolbox.com/sound-pressure-d_711.html
+      sound_pressure = np.sqrt(np.mean(self.measurements ** 2))  # RMS of amplitudes
+      sound_pressure_level = 20 * np.log10(sound_pressure / REFERENCE_SPL) if sound_pressure > 0 else 0  # dB
```

Comment:
```suggestion
      sound_pressure_level = 20 * np.log10(max(sound_pressure, REFERENCE_SPL) / REFERENCE_SPL)  # dB
```

Reply:
not sure which is better, what do you think?

Reply:
I think it's okay if dB goes negative